src/matrix/kaldi-matrix.cc

```diff
@@ -1437,9 +1439,11 @@ void Matrix<Real>::Read(std::istream & is, bool binary, bool add) {
     // Note, we never leave the while () loop before this
     // line (we return from it.)
  cleanup: // We only reach here in case of error in the while loop above.
-    delete cur_row;
+    if(cur_row != NULL)
```

Comment:
Okay, you're right. I may be overcompensating a bit, cause I treat Kaldi as something that is supposed to outlive many changes in standards and compiler versions.